### ISSUE TITLE
[Enhancement]: Cleanup color management

### ISSUE THREAD

**Issue by Caellian** [brndnmtthws/conky#2109]
@bi4k8 mentioned this in #1504 and I agree:

> I think we need to decide once and for all how to handle alpha in Conky: is it separate from colors, or included with them? We conflate ends and means in the current configuration scheme: `own_window_transparent` overrides `own_window_argb_value`, but should really be equivalent to `own_window_argb_value = 0`, except that it actually triggers a different implementation that can work in more X11 configurations. Meanwhile, color parsing does support reading alpha values, except for places where the parsed alpha is overridden by `xftalpha` or `own_window_argb_value`.
> 
> _Originally posted by @bi4k8 in https://github.com/brndnmtthws/conky/issues/1504#issuecomment-1512159257_

This is very confusing to deal with by users. As a user I'd expect a single `window_background = "color"` property that also accepts alpha, and then for conky to deal with all the implementation details.

There are however 3 different config options for this "1 concept":
- `own_window_argb_value`
- `own_window_colour`
- `own_window_transparent`

And additionally `own_window_argb_visual` that, when I forget about implementation details of X11, doesn't tell me much.

I suggest the first three to be replaced with simply `window_background_colo(u)r`, and for the `own_window_argb_visual` to be renamed to `enable_x11_true_transparency` because that communicates so much more details at a first glance:
- I know there's "fake" transparency
- I know it's only X11
  - because Wayland makes it hard to do the other (worse) thing
- I know it has (almost) nothing to do with actual "rgb" color value.

**Comment by Caellian** [comment]
When implemented, this can't break old configs, so old values should be propagated to new ones.

**Comment by Grim-GH** [comment]
It would also be nice to be able to provide additional colours beyond color0 - color9.
Ideas:
- Probably Easier? Increase to double digits (color##)
- Allow the ability to specify the colour in the conky.text
     Ex: `${alignc}${00FF00}${time %I:%M:%S %p}`
- Better but more complicated? Allow custom colour names 
      Ex. colorTime, colorCPU, colorCore1 ... colorCore16, ...etc.

**Comment by Caellian** [comment]
> - Better but more complicated? Allow custom colour names 
>       Ex. colorTime, colorCPU, colorCore1 ... colorCore16, ...etc.

You can already remove all predefined colors and add your custom ones at compile time. There's just no runtime alternative.

The way colors are handled atm is very simplistic and could be greatly improved but it requires a lot of work. The codebase is quite large and I've spent several days of my life trying to nudge some parts in the right direction only to realize whole chunks of the codebase have so many strings attached it's near impossible to incrementally improve things.

**Comment by Grim-GH** [comment]
Yeah, through viewing other conky.confs I've realized the easiest, which works for me, is using `${color #hexcode}`.
So that works for now. ;)
